Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -1130,59 +1150,61 @@ void DirectoryBlobPartition::BuildFST(std::vector<FSTBuilderNode> root_nodes, u6
   u64 total_entries = RecalculateFolderSizes(&root_nodes) + 1;
 
   const u64 name_table_offset = total_entries * ENTRY_SIZE;
-  m_fst_data.resize(name_table_offset + name_table_size);
+  std::vector<u8> fst_data(name_table_offset + name_table_size);
 
   // 32 KiB aligned start of data on disc
-  u64 current_data_address = Common::AlignUp(fst_address + m_fst_data.size(), 0x8000ull);
+  u64 current_data_address = Common::AlignUp(fst_address + fst_data.size(), 0x8000ull);
 
   u32 fst_offset = 0;   // Offset within FST data
   u32 name_offset = 0;  // Offset within name table
   u32 root_offset = 0;  // Offset of root of FST
 
   // write root entry
-  WriteEntryData(&fst_offset, DIRECTORY_ENTRY, 0, 0, total_entries, m_address_shift);
+  WriteEntryData(fst_data, &fst_offset, DIRECTORY_ENTRY, 0, 0, total_entries, m_address_shift);
 
-  WriteDirectory(&root_nodes, &fst_offset, &name_offset, &current_data_address, root_offset,
-                 name_table_offset);
+  WriteDirectory(fst_data, &root_nodes, &fst_offset, &name_offset, &current_data_address,
+                 root_offset, name_table_offset);
 
   // overflow check, compare the aligned name offset with the aligned name table size
   ASSERT(Common::AlignUp(name_offset, 1ull << m_address_shift) == name_table_size);
 
   // write FST size and location
   Write32((u32)(fst_address >> m_address_shift), 0x0424, &m_disc_header);
-  Write32((u32)(m_fst_data.size() >> m_address_shift), 0x0428, &m_disc_header);
-  Write32((u32)(m_fst_data.size() >> m_address_shift), 0x042c, &m_disc_header);
+  Write32((u32)(fst_data.size() >> m_address_shift), 0x0428, &m_disc_header);
+  Write32((u32)(fst_data.size() >> m_address_shift), 0x042c, &m_disc_header);
 
-  m_contents.AddReference(fst_address, m_fst_data);
+  m_contents.Add(fst_address, fst_data);
```

Comment:
For efficiency:

```suggestion
  m_contents.Add(fst_address, std::move(fst_data));
```

This one in particular is important, since the FST can be as big as half a megabyte for some games.